ohara-configurator/src/main/scala/com/island/ohara/configurator/route/StreamRoute.scala

```
@@ -242,121 +335,112 @@ private[configurator] object StreamRoute {
+                            log.error(ex.getMessage)
+                            StreamClusterInfo(
+                              name = data.id,
+                              imageName = STREAMAPP_IMAGE,
+                              state = Some(ContainerState.EXITED.name)
                             )
-                            res
-                          }
-                      }
-                }
-              )
+                        }
+                    }
+                    .flatMap { clusterInfo =>
+                      store.update[StreamAppDescription](
+                        id,
+                        data => Future.successful(data.copy(state = clusterInfo.state))
+                      )
+                    }
+              })
             }
           } ~
           // stop streamApp
           path(STOP_COMMAND) {
             put {
-              complete(store.value[StreamApp](id).flatMap { _ =>
-                crane.remove(id).map { _ =>
-                  store.update[StreamApp](
-                    id,
-                    d =>
-                      Future.successful(
-                        d.copy(
+              complete(
+                store.value[StreamAppDescription](id).flatMap { data =>
+                  crane
+                    .remove(data.clusterName)
+                    .map(
+                      _ =>
+                        StreamClusterInfo(
+                          name = data.clusterName,
+                          imageName = STREAMAPP_IMAGE,
                           state = None
+                      ))
+                    .recover {
+                      case ex: Throwable =>
+                        log.error(ex.getMessage)
+                        StreamClusterInfo(
+                          name = data.clusterName,
+                          imageName = STREAMAPP_IMAGE,
+                          state = Some(ContainerState.DEAD.name)
```

If the failure is caused by connection error (configurator fails to connector docker node), we still assume the container is in a **dead** state.

---

OK, I think if `stop` get some exception, we should throw it instead of swallowing it